### ISSUE TITLE
Paid Action Tests

### ISSUE THREAD

**Issue by huumn** [stackernews/stacker.news#1587]
The time has come. It'd be great to get some tests for paid actions. The don't need to be super fancy, but unit tests are not enough.

We'd probably just call the graphql api in the context of `sndev` and inspect the db/lightning nodes and make sure everything is as we expect it.

**Comment by ekzyis** [comment]
Mhh, so instead of the test suite starting all containers just for the test (and then tearing them down again), we would start everything ourselves via `sndev start` as usual and then `sndev test` would simply run commands against the containers?

Sounds good to me

**Comment by huumn** [comment]
Yep, in our case there's no reason to have the tests do orchestration other than nerd points imo.

**Comment by ekzyis** [comment]
Some cleanup would still be nice though so tests always run with the same initial data. Maybe we can at least add a test database that we then delete after the test.

**Comment by aegroto** [comment]
Just a quick question as a newbie in the project, what do you refer to as "paid action"?

Also, any idea on what frameworks you'd use to create the tests?

**Comment by huumn** [comment]
api/paidAction

**Comment by ekzyis** [comment]
> Also, any idea on what frameworks you'd use to create the tests?

we discussed above that we will probably start with no framework at all

**Comment by aniskhalfallah** [comment]
Hello, I would like to work on this !

**Comment by aegroto** [comment]
> > Also, any idea on what frameworks you'd use to create the tests?
> 
> we discussed [above](https://github.com/stackernews/stacker.news/issues/1587#issuecomment-2472121627) that we will probably start with no framework at all

Sorry for the possible misunderstanding, but isn't sndev test using jest internally?

**Comment by ekzyis** [comment]
Ah, you are right but that's for the only unit test we have and it was meant as only proof of concept

**Comment by aegroto** [comment]
> Ah, you are right but that's for the only unit test we have and it was meant as only a proof of concept

Ok great, thanks for answering. I will wait for some work to be done in this regard and try to contribute as soon as possible.
